### ISSUE TITLE
npm build errors, version 1.5.3

### ISSUE THREAD

**Issue by bbimber** [GMOD/jbrowse-components#2609]
I'm not sure if these are actually core JBrowse problems or not; however, we're trying to update our project that includes JBrowse to use 1.5.3. Among other things, we have a class that extends VcfTabixParser. Our build is complaining about a number of issues from core JBrowse classes. 

No need for you to do a lot of investigation, but do any of these make immediate sense to you? 

It's related to VcfParser / VcfTabixAdapter, and this interface, I think:

export default class extends BaseFeatureDataAdapter {
  protected configured?: Promise<{
    vcf: TabixIndexedFile
    parser: typeof VcfParser
    filehandle: GenericFilehandle
  }

The errors are along the lines of:

Property 'prototype' is missing in type 'VCF' but required in type 'typeof VCF'.
Property 'getMetadata' does not exist on type 'typeof VCF'.
Property 'parseLine' does not exist on type 'typeof VCF'.

```
ERROR in node_modules/@jbrowse/plugin-variants/src/VcfTabixAdapter/VcfTabixAdapter.ts:50:7
TS2322: Type 'Promise<{ filehandle: GenericFilehandle; vcf: TabixIndexedFile; parser: VCF; }>' is not assignable to type 'Promise<{ vcf: TabixIndexedFile; parser: typeof VCF; filehandle: GenericFilehandle; }>'.
  Type '{ filehandle: GenericFilehandle; vcf: TabixIndexedFile; parser: VcfParser; }' is not assignable to type '{ vcf: TabixIndexedFile; parser: typeof VCF; filehandle: GenericFilehandle; }'.
    Types of property 'parser' are incompatible.
      Property 'prototype' is missing in type 'VCF' but required in type 'typeof VCF'.
    48 |       })
    49 |
  > 50 |       this.configured = vcf
       |       ^^^^^^^^^^^^^^^
    51 |         .getHeader()
    52 |         .then(header => ({
    53 |           filehandle,

ERROR in node_modules/@jbrowse/plugin-variants/src/VcfTabixAdapter/VcfTabixAdapter.ts:55:33
TS2345: Argument of type '{ header: string; }' is not assignable to parameter of type '{ header: string; strict: boolean; }'.
  Property 'strict' is missing in type '{ header: string; }' but required in type '{ header: string; strict: boolean; }'.
    53 |           filehandle,
    54 |           vcf,
  > 55 |           parser: new VcfParser({ header }),
       |                                 ^^^^^^^^^^
    56 |         }))
    57 |         .catch(e => {
    58 |           this.configured = undefined

ERROR in node_modules/@jbrowse/plugin-variants/src/VcfTabixAdapter/VcfTabixAdapter.ts:77:19
TS2339: Property 'getMetadata' does not exist on type 'typeof VCF'.
    75 |   async getMetadata() {
    76 |     const { parser } = await this.configure()
  > 77 |     return parser.getMetadata()
       |                   ^^^^^^^^^^^
    78 |   }
    79 |
    80 |   public getFeatures(query: NoAssemblyRegion, opts: BaseOptions = {}) {

ERROR in node_modules/@jbrowse/plugin-variants/src/VcfTabixAdapter/VcfTabixAdapter.ts:88:31
TS2339: Property 'parseLine' does not exist on type 'typeof VCF'.
    86 |           observer.next(
    87 |             new VcfFeature({
  > 88 |               variant: parser.parseLine(line),
       |                               ^^^^^^^^^
    89 |               parser,
    90 |               id: `${this.id}-vcf-${fileOffset}`,
    91 |             }),

ERROR in src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantAdapter/ExtendedVariantAdapter.tsx:44:27
TS2352: Conversion of type 'typeof VCF' to type 'VCF' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'typeof VCF' is missing the following properties from type 'VCF': metadata, strict, samples, _parseGenotypes, and 5 more.
    42 |         const vcfConfig = await this.configure()
    43 |         const {vcf, parser} = vcfConfig
  > 44 |         const vcfParser = parser as VCF
       |                           ^^^^^^^^^^^^^
    45 |         const features : VcfFeature[] = []
    46 |
    47 |         await vcf.getLines(refName, start, end, {

webpack 5.38.1 compiled with 5 errors and 2 warnings in 17995 ms
```

**Comment by cmdcolin** [comment]
The @gmod/vcf module was coverted to typescript in it's latest edition so these are typescript specific errors. You can ignore them or try to resolve them.

An updated @gmod/vcf is now available with a fix for the "strict" error listed above (5.0.4)
Others errors may be related to using `typeof VcfParser`, might just be `VcfParser`(?)

**Comment by bbimber** [comment]
yeah, my typescript isnt that good yet, but i dont get 'typeof VcfParser' vs. just VcfParser.

**Comment by cmdcolin** [comment]
it is possible that the `typeof VcfParser` is not inside your own code. our own code was pinned to a @gmod/vcf before the typescript conversion

we fixed it here https://github.com/GMOD/jbrowse-components/pull/2595 but pending release

could try to ts-ignore the related lines in your code or pin to an older @gmod/vcf for now

for reference, the typescript playground link can help explain conceptually why it should not be typeof VcfParser https://www.typescriptlang.org/play?#code/MYGwhgzhAECyCeBhcVoG8CwAoauAOArgEYgCWw0AtvBAC4BOAXHfaQHYDmAvAOQ-YBfbMKzAA9mzrQAHo2i14eAKZiAZnCQoYXaGyUB3DckgQAFAEps4ybWjw5CY6h17DjrRexA (has errors on the one with typeof MyClass)

**Comment by bbimber** [comment]
@cmdcolin OK, thanks. I think I might need to wait out #2595. My reason to update to 1.5.3 was to take advantage of the SvgRenderer/Gylph changes and remove some hackery from our code; however, it's not critical that we do this immediately.

I am glad to see JB is moving toward typescript/stricter code, even if this is causing problems on the way.

**Comment by cmdcolin** [comment]
we published v1.5.5 which may have an improvement for this issue

**Comment by cmdcolin** [comment]
after the v1.5.5 babel misconfig, should now be fixed in v1.5.9
